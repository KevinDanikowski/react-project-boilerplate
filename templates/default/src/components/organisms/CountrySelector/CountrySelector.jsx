import React, { useState } from 'react'
import Country from '../../atoms/Country'

const CountrySelector = () => {
  // Data will be sourced from apollo once added
  const countries = [
    {
      code: 'US',
      name: 'United States'
    },
    {
      code: 'FR',
      name: 'France'
    }
  ]
  const [selectedCountryCode, handleSelectedCountryCode] = useState('US')

  return (
    <div className="country-selector">
      <div>Choose a country!</div>
      <select
        value={selectedCountryCode}
        onChange={e => handleSelectedCountryCode(e.target.value)}
        data-testid="countries"
      >
        {countries.map(country => (
          <option
            key={country.code}
            value={country.code}
            data-testid="country-option"
          >
            {country.name}
          </option>
        ))}
      </select>
      <div>Your chosen country is:</div>
      <Country
        name={countries.find(c => c.code === selectedCountryCode).name}
      />
      <style jsx>
        {`
          .country-selector {
            border: solid 1px black;
            margin: 15px;
            padding: 15px;
          }
        `}
      </style>
    </div>
  )
}

export default CountrySelector
