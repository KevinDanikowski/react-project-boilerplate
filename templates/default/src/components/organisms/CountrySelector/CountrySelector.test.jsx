import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import CountrySelector from '.'

describe('CountrySelector', () => {
  it('renders without issue', () => {
    const { asFragment } = render(<CountrySelector />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should let you choose a country', () => {
    const { getByTestId, getAllByTestId, getByDisplayValue } = render(
      <CountrySelector />
    )
    const countries = getByTestId('countries')
    const countryOptions = getAllByTestId('country-option')

    expect(countryOptions.length).toBe(2)
    expect(countries.value).toBe('US')

    fireEvent.change(getByDisplayValue(/United States/i), {
      target: { value: 'FR' }
    })

    expect(countries.value).toBe('FR')

    fireEvent.change(getByDisplayValue(/France/i), {
      target: { value: 'US' }
    })

    expect(countries.value).toBe('US')
  })
})
