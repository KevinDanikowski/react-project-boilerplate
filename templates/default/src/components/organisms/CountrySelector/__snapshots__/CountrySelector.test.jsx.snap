// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelector renders without issue 1`] = `
<DocumentFragment>
  <div
    class="jsx-2836699520 country-selector"
  >
    <div
      class="jsx-2836699520"
    >
      Choose a country!
    </div>
    <select
      class="jsx-2836699520"
      data-testid="countries"
    >
      <option
        class="jsx-2836699520"
        data-testid="country-option"
        value="US"
      >
        United States
      </option>
      <option
        class="jsx-2836699520"
        data-testid="country-option"
        value="FR"
      >
        France
      </option>
    </select>
    <div
      class="jsx-2836699520"
    >
      Your chosen country is:
    </div>
    <div
      class="jsx-563423332 country"
    >
      <div
        class="jsx-563423332"
        data-testid="country-name"
      >
        United States
      </div>
    </div>
  </div>
</DocumentFragment>
`;
